[#2701] - Add docker installation instructions and hydroshare deployment

sblack-usu opened the pull request:
<!--

Please read, and add your text at the bottom of this message.

Thanks for contributing code to HydroShare. In order to maintain code quality and expedite this process, please assist the development team by making sure the following is present in this pull request.

For more information, see https://docs.google.com/document/d/1dzxqlZW5fKNEyQSeKiSFq-SmS-VOPCva95XXkBjPExs

-->

### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

<!-- Enter steps that a QA engineer, stakeholder, or user documentation writer would follow to test the positive or "successful" case of the functionality your code provides or fixes -->

- [ ] Automated Testing

<!-- Our Jenkins Instance is set up to automatically test every commit from a pull request. Code coverage must not decrease so new functionality or code paths added during a bug fix must have appropriate tests written. Every test must pass, including PEP8 code formatting tests. -->

- [ ] Sufficient User and Developer Documentation

<!-- Please email your positive test case lbrazil@cuahsi.org, who will make the decision regarding user documentation. -->

- [ ] Passing Jenkins Build

<!-- Our Jenkins Instance is set up to automatically test every commit from a pull request. Every test must pass, including PEP8 code formatting tests. -->

- [ ] Peer Code review and approval

<!-- This is the process by which a peer developer on the HydroShare team will read the changeset, provide feedback, and ultimately give a formal approval to the code before it passes PR status. -->

### Positive Test Case
1. N/A - updated developer documentation

hydrobuild commented:
Build finished.

aphelionz commented:
@sblack-usu Can you prepare a demo for these changes for the Monday 4/30 call?

sblack-usu commented:
@aphelionz - I'm happy to.

hydrobuild commented:
Build finished.

martinseul commented on README.md:
```diff
@@ -18,19 +18,40 @@ This README file is for developers interested in working on the Hydroshare code
+    * `echo "sudo mount --bind /mnt/c /c" >> ~/.bashrc && source ~/.bashrc`
+### Configuration
+1. Clone this repository (the scripts won't work with CLRF line endings so make sure to turn off `core.autocrlf false` 
+2. `cd` into the cloned repo folder
```
I needed to cd into the Huydroshare folder and run git config core.autocrlf false. for a newby me it is not clear that this is a git command (Mark pinted this out)

martinseul commented on README.md:
```diff
@@ -18,19 +18,40 @@ This README file is for developers interested in working on the Hydroshare code
+7. bind mount within .bashrc
+    * `echo "sudo mount --bind /mnt/c /c" >> ~/.bashrc && source ~/.bashrc`
+### Configuration
+1. Clone this repository (the scripts won't work with CLRF line endings so make sure to turn off `core.autocrlf false` 
```
helpful to link here to simplify : git clone https://github.com/hydroshare/hydroshare

sblack-usu commented:
So I went on an adventure with docker on windows.  Turns out there is a documented missing feature for mounting volumes from a windows host to a linux docker container.  Basically chown commands won't work in a container on a volume hosted on a windows machine.  https://docs.docker.com/v17.09/docker-for-windows/troubleshoot/#permissions-errors-on-data-directories-for-shared-volumes

By a fluke, I did not run into this problem (and was not aware of it) until I moved my hydroshare folder when trying to organize my dev environment.  When I moved my folder, the chown commands that change /hydroshare from root:root to hydro-service:storage-hydro failed.  After some investigation and testing I discovered that I had created the folder on a shared volume within a linux vm.  Basically, because that folder structure was created on a linux vm, the chown commands worked!

This issue complicates running hydroshare on windows.  I'm happy to document how this workaround works, but I thought I'd let others know about this before updating this PR to hear what they think.  Perhaps @martinseul could try out the workaround.  I plan to continue using Docker for hydroshare as I have been.  I will notify Docker of this workaround, and perhaps they will be able to build this workaround into Docker.

aphelionz commented:
Document Virtualbox workaround and move from readme
